pilot/pkg/kube/inject/inject.go

```diff
@@ -947,3 +956,29 @@ func potentialPodName(metadata *metav1.ObjectMeta) string {
 	}
 	return ""
 }
+
+// rewriteCniPodSPec will check if values from the sidecar injector Helm
+// values need to be inserted as Pod annotations so the CNI will apply
+// the proper redirection rules.
+func rewriteCniPodSPec(annotations map[string]string, spec *SidecarInjectionSpec) error {
```

Comment:
Need some tests. Should be as simple as adding a couple yamls using cni to the testdata directory

Reply:
Starting looking to add some tests,  but I don't see an example for how to make things conditional in the current test cases based helm values.   I didn't search that hard yet but if there is another case or parameter to use as a recipe let me know.   I might not get to this for a few days due to kubecon travel.